src/app/clusters/drlc-server/drlc-server.h

```diff
@@ -0,0 +1,214 @@
+     * caller guarantees that the two structs have the same ProgramID, and the
+     * index is the index that was used to retrieve the existing struct.
+     */
+    virtual Status ReplaceLoadControlProgram(size_t aIndex,
+                                             const Structs::LoadControlProgramStruct::DecodableType & aNewProgram) = 0;
+
+    /**
+     * Add a new LoadControlProgramStruct.  The caller guarantees that there are
+     * fewer than GetNumberOfLoadControlPrograms() entries already in the list.
+     */
+    virtual Status AddLoadControlProgram(const Structs::LoadControlProgramStruct::DecodableType & aNewProgram) = 0;
+
+    /**
+     * Remove the load control program with the given id.  The caller guarantees
+     * that the id follows spec-defined constraints.
+     */
+    virtual Status RemoveLoadControlProgram(const ByteSpan & aProgramID) = 0;
+
+    /**
+     * Add a new load control event.  The caller has already done the
+     * spec-required validations.
+     */
+    virtual Status AddLoadControlEvent(const Structs::LoadControlEventStruct::DecodableType & aEvent) = 0;
+
+    /**
+     * Remove all load control events.
+     */
+    virtual Status ClearLoadControlEvents() = 0;
+};
+
+/**
+ * Instance of a DRLC cluster on a given endpoint, using the given delegate.
+ */
+template <bool ClearLoadControlEventsRequestSupported>
+class Instance : public AttributeAccessInterface, public CommandHandlerInterface
+{
+public:
+    Instance(EndpointId aEndpoint, Delegate & aDelegate);
+    ~Instance();
+
+    CHIP_ERROR Init();
+    void Shutdown();
+
+private:
+    // AttributeAccessInterface
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+    CHIP_ERROR Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder) override;
+
+    // CommandHandlerInterface
+    void InvokeCommand(HandlerContext & handlerContext) override;
+    CHIP_ERROR EnumerateAcceptedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context) override;
+
+    void HandleRegisterLoadControlProgramRequest(HandlerContext & ctx,
+                                                 const Commands::RegisterLoadControlProgramRequest::DecodableType & commandData);
+    void
+    HandleUnregisterLoadControlProgramRequest(HandlerContext & ctx,
+                                              const Commands::UnregisterLoadControlProgramRequest::DecodableType & commandData);
+    void HandleAddLoadControlEventRequest(HandlerContext & ctx,
+                                          const Commands::AddLoadControlEventRequest::DecodableType & commandData);
+    void HandleRemoveLoadControlEventRequest(HandlerContext & ctx,
+                                             const Commands::RemoveLoadControlEventRequest::DecodableType & commandData);
+    void HandleClearLoadControlEventsRequest(HandlerContext & ctx,
```

Comment:
Not sure this is in the latest spec PR: https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7762

Reply:
Yeah, looks like the PR got changed again; will take a look.